auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt

```diff
@@ -61,41 +56,27 @@ class SignInFragment : Fragment() {
                                 )
 
                                 AuthEvent.ForgotPasswordClick -> {
-                                    viewModel.forgotPasswordClickedEvent()
-                                    router.navigateToRestorePassword(parentFragmentManager)
+                                    viewModel.forgotPasswordClickedEvent(parentFragmentManager)
                                 }
 
                                 AuthEvent.RegisterClick -> {
-                                    viewModel.signUpClickedEvent()
-                                    router.navigateToSignUp(parentFragmentManager, null, null)
+                                    viewModel.signUpClickedEvent(parentFragmentManager)
                                 }
 
                                 AuthEvent.BackClick -> {
                                     requireActivity().supportFragmentManager.popBackStackImmediate()
                                 }
+
+                                is AuthEvent.OpenLink -> viewModel.openLink(
```

Comment:
```
is AuthEvent.OpenLink -> viewModel.openLink(
                                    parentFragmentManager,
                                    event.title,
                                    event.link
                                )

Reply:
we shouldn't paste logic in the UI level. So that's why all my logic is executed in the view model